modin/engines/dask/pandas_on_dask/frame/partition.py

```diff
@@ -80,25 +71,42 @@ def apply(self, func, **kwargs):
              A new `BaseFramePartition` containing the object that has had `func`
              applied to it.
         """
-        func = pkl.dumps(func)
-        call_queue = self.call_queue + [[func, kwargs]]
-        future = get_client().submit(
-            apply_list_of_funcs, call_queue, self.future, pure=False
-        )
-        futures = [get_client().submit(lambda l: l[i], future) for i in range(2)]
+        client = get_client()
+        call_queue = self.call_queue + [[func, args, kwargs]]
+        if len(call_queue) > 1:
+            future = client.submit(
+                apply_list_of_funcs, self.future, call_queue, pure=False
+            )
+        else:
+            func, args, kwargs = call_queue[0]
+            future = client.submit(apply_func, self.future, func, *args, **kwargs)
```

Comment:
We could have a function wrapper in which `if...else...` juggling occurs.

Reply:
I've forgotten the exact details as it was 28 days since previous review, but I think there should be a way to re-do it so it would be uniform regardless of single- or multi-call (without hiding the juggling - it's just sweeping stuff under the rug)

Reply:
Left as it is. See the commit ([link](https://github.com/modin-project/modin/pull/2919#discussion_r639504163)) why I had done this way. If you have any suggestions how to improve it, I would glad to hear them.